server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/aeternity/AeternityWallet.java

```diff
@@ -0,0 +1,249 @@
+    }
+
+    @Override
+    public String getCryptoAddress(String cryptoCurrency) {
+        if (!getCryptoCurrencies().contains(cryptoCurrency)) {
+            log.error("Aeternity wallet error: unknown cryptocurrency.");
+            return null;
+        }
+        return credentials.getPublicKey();
+    }
+
+    @Override
+    public BigDecimal getCryptoBalance(String cryptoCurrency) {
+        if (!getCryptoCurrencies().contains(cryptoCurrency)) {
+            log.error("Aeternity wallet error: unknown cryptocurrency.");
+            return null;
+        }
+        return rpcClient.getAccountBalance(credentials.getPublicKey());
+    }
+    
+    public BigDecimal getReceivedAmount(String address, String cryptoCurrency) {
+        if (!getCryptoCurrencies().contains(cryptoCurrency)) {
+            log.error("Aeternity wallet error: unknown cryptocurrency.");
+            return null;
+        }
+        return rpcClient.getAccountBalance(address);
+    }
+
+    
+    
+    @Override
+    public String sendCoins(String destinationAddress, BigDecimal amount, String cryptoCurrency, String description) {
+        if (!getCryptoCurrencies().contains(cryptoCurrency)) {
+            log.error("AeternityWallet wallet error: unknown cryptocurrency.");
+            return null;
+        }
+		
+		try {
+			log.info("AeternityWallet sending coins from " + credentials.getPublicKey() + " to: " + destinationAddress + " " + amount + " " + cryptoCurrency);
+			Tx signedTx = ((AeternityRPCClient)rpcClient).createAndSignTransactionAeternity(credentials.getPublicKey(), destinationAddress, amount);
+			String txid = ((AeternityRPCClient)rpcClient).sendSignedTransactionAeternity(signedTx);
+			log.debug("AeternityWallet receipt = " + txid);
+			return txid;
+		}
+		catch (Exception e) {
+			e.printStackTrace();
+		}
+		
+        return null;
+    }
+    
+    public String getWalletPrivateKey () {
+    	return credentials.getPrivateKey();
+    }
+    
+    public static AeternityWallet loadAeternityWallet(String address) {
+    	String mnemonics = null;
+    	try(BufferedReader br = new BufferedReader(new FileReader(WALLETS_PATH + address))) {
+    	    StringBuilder sb = new StringBuilder();
+    	    String line = br.readLine();
+
+    	    while (line != null) {
+    	        sb.append(line);
+    	        line = br.readLine();
+    	    }
+    	    mnemonics = sb.toString();
+    	} catch (FileNotFoundException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		} catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+    	return new AeternityWallet(mnemonics);
+    }
+    
+    //saving mnomonics in file in configured path. later will be used by AeternityRPCClient.createAndSignTransactionAeternity(String, String, BigDecimal) to initialized wallets privKey and sign the transaction.
+    //TODO: change plain text format  with some encryption after tests ? 
+    public void saveAeternityWallet() {
```

Comment:
Don't use file system to store secret stuff such as seed. In fact don't store it anywhere. keep it in memory only.

Reply:
In terms of BCH(BitcoinCore), wallet means storage for multiple key pair(pubKey + privKey), node instance can access wallet storage when signing transaction. So signrawtransaction rpc desnt need to pass privKey params for signing.

In terms of Aeternity, wallet means single key pair(pubKey + privKey), which is not stored in similar storage as BCH. Aeternity node doesnt have any access to wallet privKey

When creating TX in payment provider, senders privKey is loaded from local file (filename = pubKey) to sign the transaction. In current implemetation mnemonics are saved in plain text, for testing purposes but eventually they could be enrypted or even stored in more secure place.
Thahts why when creating new AternityWallet mnemonics are saved to local file.